.buildkite/scripts/serverless/create_deploy_tag/release_wizard_messaging.ts

```diff
@@ -0,0 +1,371 @@
+
+  const wizardSteps = Object.keys(states)
+    .filter((stateName) => states[stateName].display)
+    .map((stateName) => {
+      const stateInfo = states[stateName];
+      const stateStatus = stateData[stateName];
+      const stateEmoji = {
+        ok: ':white_check_mark:',
+        nok: ':x:',
+        pending: ':hourglass_flowing_sand:',
+        missing: ':white_circle:',
+      }[stateStatus || 'missing'];
+
+      if (stateStatus === 'pending') {
+        return `<div>[${stateEmoji}] ${stateInfo.name}<br />&nbsp; - ${stateInfo.description}</div>`;
+      } else {
+        return `<div>[${stateEmoji}] ${stateInfo.name}</div>`;
+      }
+    });
+
+  const wizardHtml = `<section>
+${wizardHeader}
+${wizardSteps.join('\n')}
+</section>`;
+
+  buildkite.setAnnotation(WIZARD_CTX_DEFAULT, 'info', wizardHtml);
+}
+
+function updateWizardInstruction(targetState: string, stateData: any) {
+  const { instructionStyle, instruction } = states[targetState];
+
+  if (instruction) {
+    buildkite.setAnnotation(
+      WIZARD_CTX_INSTRUCTION,
+      instructionStyle || 'info',
+      `<strong>${instruction}</strong>`
+    );
+  }
+}
+
+async function tryCall(fn: any, ...args: any[]) {
+  if (typeof fn === 'function') {
+    try {
+      const result = await fn(...args);
+      return result !== false;
+    } catch (error) {
+      console.error(error);
+      return false;
+    }
+  } else {
+    return true;
+  }
+}
+
+async function sendReleaseSlackAnnouncement({
+  targetCommitData,
+  currentCommitSha,
+  deployTag,
+}: {
+  targetCommitData: GithubCommitType;
+  currentCommitSha: string | undefined | null;
+  deployTag: string | undefined | null;
+}) {
+  const textBlock = (...str: string[]) => ({ type: 'mrkdwn', text: str.join('\n') });
+  const buildShortname = `kibana-serverless-release #${process.env.BUILDKITE_BUILD_NUMBER}`;
+
+  const isDryRun = process.env.DRY_RUN?.match('(1|true)');
+  const mergedAtDate = targetCommitData.commit?.committer?.date;
+  const mergedAtUtcString = mergedAtDate ? new Date(mergedAtDate).toUTCString() : 'unknown';
+  const targetCommitSha = targetCommitData.sha;
+  const targetCommitShort = targetCommitSha.slice(0, 12);
+  const compareResponse = (
+    await octokit.repos.compareCommits({
+      owner: 'elastic',
+      repo: 'kibana',
+      base: currentCommitSha || 'main',
+      head: targetCommitSha,
+    })
+  ).data;
+  const compareLink = currentCommitSha
+    ? `<${compareResponse.html_url}|${compareResponse.total_commits} commits>`
+    : 'a new release candidate';
+
+  const mainMessage = [
+    `:ship_it_parrot: Promotion of a new ${compareLink} has been <${process.env.BUILDKITE_BUILD_URL}|initiated>!\n`,
+    `*Remember:* Promotion to Staging is currently a manual process and will proceed once the build is signed off in QA.\n`,
+  ];
+  if (isDryRun) {
+    mainMessage.unshift(
+      `*:memo:This is a dry run - no commit will actually be promoted. Please ignore!*\n`
```

Comment:
I don't think we need the trailing line break?

```suggestion
      `*:memo:This is a dry run - no commit will actually be promoted. Please ignore!*`
```

Reply:
This is to ensure the 1 empty line between paragraphs as you suggested for other blocks of the main message.

Reply:
Each `section` block automatically has spacing between them ([example](https://app.slack.com/block-kit-builder/T0CUZ52US#%7B%22blocks%22:%5B%7B%22type%22:%22section%22,%22text%22:%7B%22type%22:%22mrkdwn%22,%22text%22:%22Section%201,%20line%201%5CnSection%201,%20line%202%22%7D%7D,%7B%22type%22:%22section%22,%22text%22:%7B%22type%22:%22mrkdwn%22,%22text%22:%22Section%202,%20line%201%5CnSection%202,%20line%202%22%7D%7D%5D%7D))

Reply:
Doh! I misread this as a `push` and not an `unshift`.. just ignore me 😅